### ISSUE TITLE
Gateway-Monitoring CARP IPv6

### ISSUE THREAD

**Issue by tuxnet** [opnsense/core#5824]
**Important notices**

Before you add a new report, we ask you kindly to acknowledge the following:

- [x] I have read the contributing guide lines at https://github.com/opnsense/core/blob/master/CONTRIBUTING.md
- [x] I am convinced that my issue is new after having checked both open and closed issues at https://github.com/opnsense/core/issues?q=is%3Aissue

**Describe the bug**

Since the upgrade to 22.1.9, gateway monitoring no longer works properly for IPv6 gateways.
The dpinger process is started with -B \<IPv6 CARP\> and not with -B \<IPv6 Interface-Address\>.
As a result, IPv6 gateway monitoring only works on the CARP active firewall.

**To Reproduce**

1. Configure CARP on the WAN interface + gateway for two firewalls
2. Accessing the backup firewall with ssh
3. execute the command 'ps -ax | grep dpinger'.
4. Check the IPv6 address in the command of the process (after -B)

**Expected behavior**

The dpinger process should be started with -B \<IPv6 Interface-Address\>.

**Describe alternatives you considered**

The monitoring source can be selected

**Environment**

OPNsense 22.1.9-amd64

**Comment by fichtner** [comment]
I keep looking at the code and wonder why this worked at all in the past being so poorly modelled in terms of which address should be used when. 22.1.9 fixed a case where IPv6 was ignored when primary address was set to none. We can traverse the primary IP first but at some point this breaks user expectations just like IPv4 where the first closest match on the subnet wins. I'll propose a patch soon.

Cheers,
Franco

**Comment by fichtner** [comment]
Does this work for you?

    # opnsense-patch 09ff74b774


Cheers,
Franco

**Comment by tuxnet** [comment]
Yes, it works. For active and backup now the interface and not the carp ip is used.
Thanks for the quick patch!!

kind regards
